### ISSUE TITLE
Add symmetric pad mode in addition to reflect

### ISSUE THREAD

**Issue by TomWildenhain-Microsoft** [onnx/onnx#3695]
# Feature Request

### What is the problem that this feature solves?
Tensorflow has a SYMMATRIC padding mode similar to REFLECT that duplicates the edge values

### Describe the alternatives you have considered
Pad using SYMMETRIC can be simulated with pad using CONSTANT (of amount 1), then pad with REFLECT (original amount), then COMPRESS along every axis to remove extra row.

### Describe the feature
TF has this mode.  People need it sometimes (2 requests in tf2onnx).
https://github.com/onnx/tensorflow-onnx/issues/1686
https://github.com/onnx/tensorflow-onnx/issues/72

### Will this influence the current api?
Yes, add REFLECT mode to pad.

**Comment by askhade** [comment]
@TomWildenhain-Microsoft : Can you send a PR for this change?

**Comment by svebert** [comment]
Hi! 
I am missing also the pad mode "SYMMETRIC". 
The tf.pad function has three modes: 'CONSTANT', 'REFLECT' and 'SYMMETRIC'. The latter is not supported by onnx.
I created a workaround for padding values not greater than 1. I know, its not much and my code is probably ugly :). Maybe someone can use it.
The first class `SymmetricPadding2D `is a "regular" symmetric padding layer, which will work in Keras/Tensorflow
The second class `SymmetricPadding2DONNX `works for `tf2onnx.convert.from_keras`. Just replace in your keras model `SymmetricPadding2D `with `SymmetricPadding2DONNX`.

(Its only working for padding up to 1 pixel in each direction)

```python
from tensorflow.keras.layers import InputSpec, Layer
import tensorflow as tf

class SymmetricPadding2D(Layer):
    def __init__(self, padding=(1, 1), **kwargs):
        super(SymmetricPadding2D, self).__init__(**kwargs)
        self.padding = tuple(padding)
        self.input_spec = [InputSpec(ndim=4)]

    def compute_output_shape(self, s):
        return (
            s[0] + self.padding[0][0], s[1] + self.padding[0][1], s[2] + self.padding[1][0], s[3] + self.padding[1][1])

    def padding_values(self):
        w_pad, h_pad = self.padding
        if isinstance(w_pad, int):
            w1_pad = w_pad
            w2_pad = w_pad
        elif isinstance(w_pad, tuple) and len(w_pad) > 1:
            w1_pad = w_pad[0]
            w2_pad = w_pad[1]
        else:
            raise ValueError("Unsupported padding")

        if isinstance(h_pad, int):
            h1_pad = h_pad
            h2_pad = h_pad
        elif isinstance(h_pad, tuple) and len(h_pad) > 1:
            h1_pad = h_pad[0]
            h2_pad = h_pad[1]
        else:
            raise ValueError("Unsupported padding")

        return h1_pad, h2_pad, w1_pad, w2_pad

    def call(self, x, *args, **kwargs):
        h1_pad, h2_pad, w1_pad, w2_pad = self.padding_values()
        return tf.pad(x, [[0, 0], [h1_pad, h2_pad], [w1_pad, w2_pad], [0, 0]], 'SYMMETRIC')


def duplicate_last_row(x):
    return tf.concat((x, tf.expand_dims(x[:, -1, ...], 1)), axis=1)


def duplicate_last_col(x):
    return tf.concat((x, tf.expand_dims(x[:, :, -1, ...], 2)), axis=2)


def duplicate_first_row(x):
    return tf.concat((tf.expand_dims(x[:, 0, ...], 1), x), axis=1)


def duplicate_first_col(x):
    return tf.concat((tf.expand_dims(x[:, :, 0, ...], 2), x), axis=2)


def apply_symmetric_part(x, h1_pad, h2_pad, w1_pad, w2_pad):
    if h1_pad > 0:
        x = duplicate_first_row(x)
    if h2_pad > 0:
        x = duplicate_last_row(x)
    if w1_pad > 0:
        x = duplicate_first_col(x)
    if w2_pad > 0:
        x = duplicate_last_col(x)
    return x


class SymmetricPadding2DONNX(SymmetricPadding2D):
    def __init__(self, padding=(1, 1), **kwargs):
        super().__init__(padding, **kwargs)

    def call(self, x, *args, **kwargs):
        h1_pad, w1_pad, h2_pad, w2_pad = self.padding_values()
        if w1_pad > 1 or w2_pad > 1 or h1_pad > 1 or h2_pad >1:
            raise ValueError("padding value must not exceed 1")
        return apply_symmetric_part(x, h1_pad, w1_pad, h2_pad, w2_pad)
```

**Comment by fdwr** [comment]
> github-actions closed this as completed on Jan 5, 2023

I don't see "symmetric" in https://onnx.ai/onnx/operators/onnx__Pad.html. Was this *actually completed*, or should closure reason have been *not planned*?
